pkgs/build-support/testers/hasPkgConfigModules/tester.nix

```diff
@@ -45,7 +45,7 @@ runCommand testName {
         if [[ "$moduleVersion" == "$version" ]]; then
           echo "✅ pkg-config module $moduleName exists and has version $moduleVersion"
         else
-          echo "❌ pkg-config module $moduleName exists and has version $moduleVersion when $version was expected"
+          echo "${if versionCheck then "❌" else "⚠"} pkg-config module $moduleName exists and has version $moduleVersion when $version was expected"
           ((versionMismatch+=1))
```

Comment:
To make my suggestion more concrete, from https://github.com/NixOS/nixpkgs/pull/311069#issuecomment-2106355589

This is a little silly and untested, but I guess it gets the intended behavior across, fwiw.

```suggestion
          ${if versionCheck
            then ''
              echo "❌ pkg-config module $moduleName exists and has version $moduleVersion when $version was expected"
              ((versionMismatch+=1))
            ''
            # TODO: figure out the UX for enabling the version check; see https://github.com/NixOS/nixpkgs/pull/311069
            else ''
              echo "✅ pkg-config module $moduleName exists and has version $moduleVersion"
            ''}
```

Reply:
It's tempting to add more info to the message, but that will just serve to distract whoever reads it, if we don't explain it, which is out of scope for this PR.